public/components/ui_action_context_menu.tsx

```diff
@@ -39,6 +42,35 @@ export const ActionContextMenu = (props: Props) => {
   });
   const resultSummaryEnabled = useObservable(props.resultSummaryEnabled$, false);
   const isQuerySummaryCollapsed = useObservable(props.isQuerySummaryCollapsed$, false);
+  const isSummaryAgentAvailable = useObservable(props.isSummaryAgentAvailable$, false);
+
+  async function checkAgentsExist(
+    http: HttpSetup,
+    agentConfigName: string | string[],
+    dataSourceId?: string
+  ) {
+    const response = await http.get(AGENT_API.CONFIG_EXISTS, {
+      query: { agentConfigName, dataSourceId },
+    });
+    return response;
+  }
+
+  useEffect(() => {
+    const fetchSummaryAgent = async () => {
+      try {
+        const summaryAgentStatus = await checkAgentsExist(
+          props.httpSetup,
+          DATA2SUMMARY_AGENT_CONFIG_ID,
+          props.data.query.queryString.getQuery().dataset?.dataSource?.id
```

Comment:
We should listen to dataSourceId change. And if the resultSummaryEnabled is false, we do not need to check if agents exists.